lightning-c-bindings/src/chain/chaininterface.rs

```diff
@@ -0,0 +1,140 @@
+impl lnBroadcasterInterface for BroadcasterInterface {
+	fn broadcast_transaction(&self, tx: &bitcoin::blockdata::transaction::Transaction) {
+		let c_tx = ::bitcoin::consensus::encode::serialize(tx);
+		(self.broadcast_transaction)(self.this_arg, crate::c_types::Transaction::from_slice(&c_tx))
+	}
+}
+
+// We're essentially a pointer already, or at least a set of pointers, so allow us to be used
+// directly as a Deref trait in higher-level structs:
+impl std::ops::Deref for BroadcasterInterface {
+	type Target = Self;
+	fn deref(&self) -> &Self {
+		self
+	}
+}
+/// " An enum that represents the speed at which we want a transaction to confirm used for feerate"
+/// " estimation."
+#[repr(C)]
+pub enum ConfirmationTarget {
+	/// " We are happy with this transaction confirming slowly when feerate drops some."
+	Background,
+	/// " We'd like this transaction to confirm without major delay, but 12-18 blocks is fine."
+	Normal,
+	/// " We'd like this transaction to confirm in the next few blocks."
+	HighPriority,
+}
+use lightning::chain::chaininterface::ConfirmationTarget as lnConfirmationTarget;
+impl ConfirmationTarget {
+	#[allow(unused)]
+	pub(crate) fn to_ln(&self) -> lnConfirmationTarget {
+		match self {
+			ConfirmationTarget::Background => lnConfirmationTarget::Background,
+			ConfirmationTarget::Normal => lnConfirmationTarget::Normal,
+			ConfirmationTarget::HighPriority => lnConfirmationTarget::HighPriority,
+		}
+	}
+	#[allow(unused)]
+	pub(crate) fn from_ln(lnt: lnConfirmationTarget) -> Self {
+		match lnt {
+			lnConfirmationTarget::Background => ConfirmationTarget::Background,
+			lnConfirmationTarget::Normal => ConfirmationTarget::Normal,
+			lnConfirmationTarget::HighPriority => ConfirmationTarget::HighPriority,
+		}
+	}
+}
+/// " A trait which should be implemented to provide feerate information on a number of time"
+/// " horizons."
+/// ""
+/// " Note that all of the functions implemented here *must* be reentrant-safe (obviously - they're"
+/// " called from inside the library in response to ChainListener events, P2P events, or timer"
+/// " events)."
+#[repr(C)]
+pub struct FeeEstimator {
+	pub this_arg: *mut c_void,
+	/// " Gets estimated satoshis of fee required per 1000 Weight-Units."
+	/// ""
+	/// " Must be no smaller than 253 (ie 1 satoshi-per-byte rounded up to ensure later round-downs"
+	/// " don't put us below 1 satoshi-per-byte)."
+	/// ""
+	/// " This translates to:"
+	/// "  * satoshis-per-byte * 250"
+	/// "  * ceil(satoshis-per-kbyte / 4)"
+	pub get_est_sat_per_1000_weight: extern "C" fn (this_arg: *const c_void, confirmation_target: ConfirmationTarget) -> u64,
+}
+unsafe impl Sync for FeeEstimator {}
+unsafe impl Send for FeeEstimator {}
+
+use lightning::chain::chaininterface::FeeEstimator as lnFeeEstimator;
+impl lnFeeEstimator for FeeEstimator {
+	fn get_est_sat_per_1000_weight(&self, confirmation_target: lightning::chain::chaininterface::ConfirmationTarget) -> u64 {
```

Comment:
I think in this case its because its defined in this file. There are a few cases where it isnt. While maybe we could add imports, its probably not worth the effort.